stock_inventory_valuation_report/reports/stock_inventory_valuation_report.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class StockInventoryValuationView(models.TransientModel):
+    _name = "stock.inventory.valuation.view"
+    _description = "Stock Inventory Valuation View"
+
+    report_id = fields.Many2one("report.stock.inventory.valuation.report")
+
+    name = fields.Char()
+    reference = fields.Char()
+    barcode = fields.Char()
+    qty_at_date = fields.Float()
+    uom_id = fields.Many2one(
+        comodel_name="uom.uom",
+    )
+    currency_id = fields.Many2one(
+        comodel_name="res.currency",
+    )
+    cost_currency_id = fields.Many2one(
+        comodel_name="res.currency",
+    )
+    standard_price = fields.Float()
+    stock_value = fields.Float()
+    cost_method = fields.Char()
+
+
+class StockInventoryValuationReport(models.TransientModel):
+    _name = "report.stock.inventory.valuation.report"
+    _description = "Stock Inventory Valuation Report"
+
+    # Filters fields, used for data computation
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+    )
+    inventory_datetime = fields.Datetime(required=True, default=fields.Datetime.now)
+
+    results = fields.One2many(
+        comodel_name="stock.inventory.valuation.view",
+        inverse_name="report_id",
+        compute="_compute_results",
+    )
+
+    @api.depends("inventory_datetime")
+    def _compute_results(self):
+        self.ensure_one()
+        domain = [("type", "=", "product")]
+        product_id = self.env.context.get("product_id", False)
+        product_tmpl_id = self.env.context.get("product_tmpl_id", False)
+        if product_id:
+            domain = expression.AND([domain, [("id", "=", product_id)]])
+        elif product_tmpl_id:
+            domain = expression.AND(
+                [domain, [("product_tmpl_id", "=", product_tmpl_id)]]
+            )
+        products = (
+            self.env["product.product"]
+            .with_context(
+                to_date=self.inventory_datetime,
+                company_owned=True,
+                create=False,
+                edit=False,
+            )
+            .search(domain)
+            # 'quantity_svl' we can't use compute field for search
+        ).filtered(lambda pp: pp.quantity_svl != 0)
+        results = self.env["stock.inventory.valuation.view"]
+        if products:
+            for product in products:
```

Comment:
Here you could use `new` function once. In first step you prepare vals and second step `new` function.
```py
vals = [{....} for product in products]
results = resultnts.new(vals)
```

Reply:
`.new()` cannot take a list of dictionaries, only a dictionary. I'll just use `.create()` instead.